fiftyone/operators/executor.py

```diff
@@ -468,13 +494,20 @@ def dataset_id(self):
     @property
     def view(self):
         """The :class:`fiftyone.core.view.DatasetView` being operated on."""
-        if self._view is not None:
+        # since the context.dataset may have been changed, we should only
+        # return the current view if the current DatasetView's root dataset
+        # matches the dataset in the context
+        if self._view and self._view._root_dataset.name == self.dataset.name:
             return self._view
 
-        # Forces a dataset reload if necessary
-        _ = self.dataset
+        # Always derive the view from the context's dataset
+        dataset = self.dataset
```

Comment:
You could avoid needing any changes or complex logic in the `view` property if you just update `request_params["dataset_name"]` within the `dataset` property, if the name changed.

Line 504 ensures that `dataset` is accessed before `get_view()` is called

Reply:
that is intentional to ensure the view matches the dataset... it follows the pattern the app uses to keep the state.view compatible with state.dataset

Reply:
In this class, `self.dataset` does the work of ensuring that the dataset is properly reloaded (including loading it under it's current name, if it has changed), so the only change needed in this function is to do this:

```py
dataset_name = self.dataset.name
```

instead of referencing `request_params["dataset_name"]` as was previously done (your idea to use `dataset.name` is good here 👍)